front/lib/actions/mcp_internal_actions/servers/file_generation.ts

```diff
@@ -251,6 +271,145 @@ const createServer = (auth: Authenticator): McpServer => {
         };
       }
 
+      // If the format requires conversion and we have plain text content,
+      // we need to convert it to HTML first.
+      if (
+        isBinaryFormat(extension) &&
+        !validateUrl(file_content).valid &&
+        !getResourceNameAndIdFromSId(file_content)
+      ) {
+        const convertapi = new ConvertAPI(process.env.CONVERTAPI_API_KEY);
+
+        try {
+          let htmlContent: string;
+
+          // Convert content to HTML based on source format
+          switch (source_format) {
+            case "markdown": {
+              // Parse markdown to HTML
+              const parsedMarkdown = await marked.parse(file_content);
```

Comment:
I need to improve this part but merging already to fix the bug, will open another PR for improvements.